test/utils.js

```diff
@@ -254,4 +254,10 @@ describe('utils', function () {
       assert.deepEqual(utils.validateProperties(properties), expected);
     });
   });
+
+  describe('isWebWorkerEnvironment', function () {
```

Comment:
It would be nice to have a couple tests for the WebWorker environment as well. Not sure if they *work* but seems there are a couple plugins for Jest:
- https://www.npmjs.com/package/jsdom-worker
- https://www.npmjs.com/package/jest-worker

Reply:
Since we use karma as test runner with mocha + chai, I thought of finding a framework that's compatible with karma. I found [karma-mocha-webworker](https://www.npmjs.com/package/karma-mocha-webworker). Though the package is pretty old, I found a way to make it work. If we didn't have karma, I'd totally go with jest. I pushed these changes and should run as part of `make test`.